baselines/diabetic_retinopathy_detection/radial.py

```diff
@@ -0,0 +1,356 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""ResNet50 Radial BNN trained on Kaggle's Diabetic Retinopathy Detection.
+
+This script performs variational inference with a Radial posterior,
+as introduced in Radial Bayesian Neural Networks (Farquhar et al. 2020).
+
+We use a few additional, noteworthy techniques in training:
+
+1. Normal (or Radial) prior whose mean is tied at the variational posterior's.
+    This makes the KL penalty only penalize the weight posterior's standard
+    deviation and not its mean.
+    The prior's standard deviation is a fixed hyperparameter.
+2. Flipout for lower-variance gradients in convolutional layers and the final
+    dense layer (Wen et al., 2018).
+3. KL annealing (Bowman et al., 2015).
+"""
+
+import os
+import time
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from absl import app
+from absl import flags
+from absl import logging
+
+import uncertainty_baselines as ub
+import utils  # local file import
+
+DEFAULT_BATCH_SIZE = 16
+DEFAULT_NUM_EPOCHS = 90
+
+# Data load / output flags.
+flags.DEFINE_string(
+    'output_dir', '/tmp/diabetic_retinopathy_detection/radial',
+    'The directory where the model weights and '
+    'training/evaluation summaries are stored.')
+flags.DEFINE_string('data_dir', None, 'Path to training and testing data.')
+flags.mark_flag_as_required('data_dir')
+
+# Learning rate / SGD flags.
+flags.DEFINE_float(
+  'base_learning_rate', 4e-4,
+  'Base learning rate when total batch size is DEFAULT_TRAIN_BATCH_SIZE. It is '
+  'scaled by the ratio of the total batch size to DEFAULT_TRAIN_BATCH_SIZE.')
+flags.DEFINE_integer(
+  'lr_warmup_epochs', 1,
+  'Number of epochs for a linear warmup to the initial '
+  'learning rate. Use 0 to do no warmup.')
+flags.DEFINE_float('lr_decay_ratio', 0.2, 'Amount to decay learning rate.')
+flags.DEFINE_list(
+  'lr_decay_epochs', ['30', '60'],
+  'Epochs to decay learning rate by.')
+
+# Radial BNN flags.
+flags.DEFINE_bool(
+  'tied_mean_prior', True,
+  'If True, fix the mean of the prior to that of the variational posterior. '
+  'This causes the KL to only penalize the standard deviation of the weight '
+  'posterior, and not its mean.')
+flags.DEFINE_bool(
+  'radial_prior', False,
+  'If True, use a Radial prior instead of a Normal. This is by default False '
+  'to enable comparability to other methods, but can be helpful in e.g. '
+  'variational continual learning (Nguyen et al. 2018, Farquhar et al. 2020).')
+flags.DEFINE_integer(
+  'kl_annealing_epochs', 200,
+  'Number of epochs over which to anneal the KL term to 1.')
+flags.DEFINE_float('prior_stddev', 0.1, 'Fixed stddev for weight prior.')
+flags.DEFINE_float(
+  'stddev_init', 1e-3,
+  'Initialization of posterior standard deviation parameters.')
+
+# General model flags.
+flags.DEFINE_integer('seed', 42, 'Random seed.')
+flags.DEFINE_float('l2', 5e-5, 'L2 regularization coefficient.')
```

Comment:
Why use an L2 reg? The ELBO ought to serve that purpose, at least if you're not doing the tying to the mean thing. I'm very confused why one would compute the KL to a tied prior and then separately regularize the mean values back to zero?